src/date-picker/date-picker.tsx

```diff
@@ -0,0 +1,1311 @@
+import React, {
```

Comment:
This component is too big already. Can you consider pulling the calendar out instead?